content/reference/v1beta1/repeatable.md

```diff
@@ -0,0 +1,278 @@
+
+**Repeatable Items presently only work for text and textarea types**
+
+To make an item repeatable, set `repeatable` to true
+```yaml
+    - name: ports
+      items:
+      - name: service_ports
+        title: Service Ports
+        type: text
+        repeatable: true
+```
+
+Repeatable items do not use the `default` or `value` fields, but instead a `valuesByGroup` field.
+`valuesByGroup` should have an entry for the parent Config Group name, with all default `key:value` pairs nested in the group.  1 default entry is required for the repeatable item.
+```yaml
+    valuesByGroup:
+      ports:
+        port-default-1: "80"
+```
+
+## Template Targets
+
+Repeatable items require at least 1 `template` to be provided.  The `template` defines a YAML target in the manifest to duplicate for each repeatable item.
+
+The entire YAML node at the target will be duplicated, including nested fields.
+
+The `yamlPath` field of the `template` must denote index position for arrays using square brackets.  For example, `spec.ports[0]` to select the first port entry for duplication.  All duplicate YAML will be appended to the final array in the `yamlPath`.
+
+`yamlPath` **must** end with an array.
+```yaml
+    template:
+    - apiVersion: v1
+      kind: Service
+      name: my-service
+      namespace: my-app
+      yamlPath: 'spec.ports[0]'
+```
+
+If the `yamlPath` field is blank, the entire YAML document matching the `template` will be replaced with a copy for each of the repeatable item entries.  The `metadata.name` field of the new doc will reflect the repeatable item `key`.
+
+## Templating
+
+The repeat items are called with the delimeters `repl[[ .itemName ]]` or `[[repl .itemName ]]`.  These delimiters can be placed anywhere inside of the `yamlPath` target node.
+```yaml
+    - port: '{{repl ConfigOption "[[repl .service_ports ]]" | ParseInt }}'
+```
+This repeatable templating is not compatible with sprig templating functions.  It is designed for inserting repeatable `keys` into the manifest. Repeatable templating **can** be placed inside of Replicated config templating.
+
+## Ordering
+
+Repeatable templates are processed before config template rendering.
+
+Repeatable items are processed in order of the template targets in the Config Spec file.  Effectively, this ordering is from the top of the Config Spec, by Config Group, by Config Item, and then by template target.
+
+```yaml
+    - name: ports
+      items:
+      - name: service_ports
+        title: Service Ports
+        type: text
+        repeatable: true
+        template:
+        - apiVersion: v1 #processed first
+          kind: Service
+          name: my-service
+          namespace: my-app
+          yamlPath: 'spec.ports[0]'
+        - apiVersion: v1 #processed second
+          kind: Service
+          name: my-service
+          namespace: my-app
+          yamlPath:
+    {...}
```

Comment:
I've added some annotation to the ellipses.  The first one is cutting out everything until the next item, and the second one is cutting out everything until the next group.  I agree it looks kinda clunky though, if you have any thoughts on splitting this better I'm all for it.

Reply:
Yeah, I see my confusion now. Can the first {...} be indented two more spaces so that it aligns with the items, since that's what it is representing? And then maybe say "other items" and "other deployments" or something like that since there could be multiple.

Reply:
Updated, let me know if this makes more sense

Reply:
Looking at it now, the only possible concern is that the ordering isn't correct anymore if there are other items concealed by the ellipses. Like the fourth one wouldn't really be fourth if there are other items in `ports`. Sorry for waffling on this one, but maybe we can remove the ellipses because they confuse the ordering a bit. People will probably get that there can be other things in there since that's the point of repeatable stuff.

Reply:
the real purpose of the ellipses is that repeatable items are required to have the ValuesByGroup property filled in, which would make this example much longer than it needs to be.  The elipses with the note "other item properties" is supposed to denote that the item is abbreviated for space, not that there are other repeatable items present but not listed.  I wouldn't think anybody would assume there are other repeatable items listed inside of the ellipses, since the example is meant to show all of the repeatable items and when they'll be processed.

Reply:
Ah I'm with you now. You can resolve this one too then.